eevee/metrics/classification.py

```diff
@@ -96,3 +96,108 @@ def intent_report(
             weighted_group_df.set_index('group', inplace=True)
             
             return weighted_group_df
+
+
+def intent_layers_report(
+        true_labels: pd.DataFrame,
+        pred_labels: pd.DataFrame,
+        intent_layers: Optional[Dict[str, Dict[str, List[str]]]] = None,
+        breakdown=False,
+):
+    df = pd.merge(true_labels, pred_labels, on="id", how="inner")
+
+    # for cases where we are seeing NaN values popping up.
+    df[['intent_x', 'intent_y']] = df[['intent_x', 'intent_y']].fillna(value="_")
+
+    # aliasing preds
+    col = "intent_y"
+    intents_dict = {value: key for key, values in intent_layers.get(col).items() for value in values}
+    df[col] = df[col].apply(lambda intent: intents_dict.get(intent, intent))
+
+    #aliasing trues
+    col = "intent_x"
+    intents_dict = {value: key for key, values in intent_layers.get(col).items() for value in values}
+    df["{}-alias".format(col)] = df[col].apply(lambda tag: intents_dict.get(col).get(tag, tag))
```

Comment:
it is crashing here:
```
    df["{}-alias".format(col)] = df[col].apply(lambda tag: intents_dict.get(col).get(tag, tag))
AttributeError: 'NoneType' object has no attribute 'get'
```
where `col` is `intent_x` and `intents_dict` looks like this:

```
{'_': 'acoustic_oos',
 '_oos_': 'lexical_oos',
 'audio_channel_noise': 'acoustic_oos',
 'audio_channel_noise_hold': 'acoustic_oos',
 'audio_silent': 'acoustic_oos',
 'audio_speech_unclear': 'acoustic_oos',
 'audio_speech_volume': 'acoustic_oos',
 'background_noise': 'acoustic_oos',
 'background_speech': 'acoustic_oos',
 'ood': 'lexical_oos',
 'other_language': 'acoustic_oos',
 'partial': 'lexical_oos'}
 ```

Reply:
fixed